api/models/instances.go

```diff
@@ -12,6 +12,18 @@ import (
 )
 
 func InstanceKeyroll() error {
+	system, err := Provider().SystemGet()
+	if err != nil {
+		return err
+	}
+
+	// only allow running and converging status through
+	switch system.Status {
+	case "running", "converging":
+	default:
+		return fmt.Errorf("unable to keyroll with rack status: %s", system.Status)
```

Comment:
After reading it 100x, think you mean the error message should state that. Still waking up...